test/meson.build

```diff
@@ -0,0 +1,74 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+gmock_main_dep = dependency('gmock_main')
+
+conf_data = configuration_data()
+conf_data.set('ICEBERG_TEST_RESOURCES', meson.current_source_dir() / 'resources')
+configure_file(
+    input: 'test_config.h.in',
+    output: 'test_config.h',
+    configuration: conf_data,
+    install: true,
+    install_dir: get_option('includedir') / 'iceberg/test',
+)
+
+iceberg_tests = {
+    'schema_test': files(
+        'name_mapping_test.cc',
+        'partition_field_test.cc',
+        'partition_spec_test.cc',
+        'schema_field_test.cc',
+        'schema_test.cc',
+        'schema_util_test.cc',
+        'snapshot_test.cc',
+        'sort_field_test.cc',
+        'sort_order_test.cc',
+        'transform_test.cc',
+        'type_test.cc',
+    ),
+    'table_test': files(
+        'json_internal_test.cc',
+        'schema_json_test.cc',
+        'table_test.cc',
+        'test_common.cc',
+    ),
+    'expression_test': files('expression_test.cc', 'literal_test.cc'),
+    'json_serde_test': files(
+        'json_internal_test.cc',
+        'metadata_serde_test.cc',
+        'schema_json_test.cc',
+        'test_common.cc',
+    ),
+    'util_test': files(
+        'config_test.cc',
+        'decimal_test.cc',
+        'endian_test.cc',
+        'formatter_test.cc',
+        'string_util_test.cc',
+        'visit_type_test.cc',
+    ),
+}
+
+foreach test_name, sources : iceberg_tests
```

Comment:
I'm just a little bit worried that it may add more maintenance overhead to contributors who don't know Meson before. In addition, people may forget to update this file when adding a new test case to the CMakeLists.txt. Any best practice for this?

Reply:
It depends on how much we want to support Meson. Arrow, which doesn't officially support it, has an "Extra" CI job for Meson that only runs when tagged. It's not a critical CI job. 

Smaller projects like arrow-adbc and nanoarrow have just included Meson as part of the main CI. Given they are smaller projects it hasn't been a huge issue, and Meson arguably makes it much easier to set up valgrind, sanitizer, and coverage jobs than CMake. 

I'd suggest starting with the adbc/nanoarrow approach given this config is pretty small at the moment and theoretically not that hard to maintain, but either approach is workable